### DISCUSSION TITLE
Why a trait HasFactory and not macro's?

### DISCUSSION THREAD

**Discussion by henzeb** [worksome/request-factories#25] [Ideas]
One of the caveats of having the 'fake' method called on the FormRequest is the need for having the package installed in production. I was baffled, as FormRequest extends Request and Request is Macroable. FormRequests are Macroable. 

Is it an idea to change this into macros? That way, when the package is installed (which usually is in dev) FormRequest automatically will have the fake method.

This will have no breaking change whatsoever. Users who use the HasFactory trait can keep using it or refactor it away at their choosing.

**Comment by lukeraymonddowning** [comment]
This might actually be a very good idea. Good thinking. Would you be up for creating a PR?

**Comment by lukeraymonddowning** [comment]
We've just release v3, and moved using macros for this. Thanks for the suggestion!
